src/main/java/org/elasticsearch/common/lucene/Lucene.java

```diff
@@ -633,6 +633,30 @@ public static final boolean indexExists(final Directory directory) throws IOExce
     }
 
     /**
+     * Wait for an index to exist for up to {@code timeLimitMillis}. Returns
```

Comment:
nit pick, can we have a dedicated test for this class ? simple as the one in the `ShadowEngineTests`